ush/xml_creator.py

```diff
@@ -0,0 +1,267 @@
+'''
+A standalone tool that reads a YAML file and generates a Rocoto XML.
+'''
+
+
+import argparse
+from copy import deepcopy
+from io import StringIO
+import os
+import sys
+from textwrap import dedent
+import xml.etree.ElementTree as ET
+
+import yaml
+
+
+def load_config(arg):
+
+    '''
+    Check to ensure that the provided config file exists. If it does, load it
+    with YAML's safe loader and return the resulting dict.
+    '''
+
+    # Check for existence of file
+    if not os.path.exists(arg):
+        msg = f'{arg} does not exist!'
+        raise argparse.ArgumentTypeError(msg)
+
+    with open(arg, 'r') as fn:
+        cfg = yaml.load(fn, Loader=yaml.SafeLoader)
+
+    return cfg
+
+def path_ok(arg):
+
+    '''
+    Check whether the path to the file exists, and is writeable. Return the path
+    if it passes all checks, otherwise raise an error.
+    '''
+
+    # Get the absolute path provided by arg
+    dir_name = os.path.abspath(os.path.dirname(arg))
+
+    # Ensure the arg path exists, and is writable. Raise error if not.
+    if os.path.lexists(dir_name) and os.access(dir_name, os.W_OK):
+        return arg
+
+    msg = f'{arg} is not a writable path!'
+    raise argparse.ArgumentTypeError(msg)
+
+def cycstr(loader, node):
+
+    ''' Returns a string that has Rocoto's cyclestr tags surrounding the
+    input argument'''
+
+    arg = loader.construct_mapping(node, deep=True)
+    string = arg.pop('value')
+    attrs = ' '.join([f'{key}={value}' for key, value in arg.items()])
+    cyc = ET.Element('cyclestr', attrib=arg)
+    cyc.text = string
+    return cyc
+
+def include(loader, node):
+
+    ''' Returns a YAML config contained in a separate file. '''
+
+    filenames = loader.construct_sequence(node)
+
+    cfg = {}
+    for filename in filenames:
+        with open(filename, 'r') as fn:
+           cfg.update(yaml.load(fn, Loader=yaml.SafeLoader))
+    return cfg
+
+def startstopfreq(loader, node):
+
+    ''' Returns a Rocoto-formated string for the contents of a cycledef
+    tag '''
+
+    arg = loader.construct_sequence(node)
+    return ' '.join([str(i) for i in arg])
+
+
+yaml.add_constructor('!cycstr', cycstr, Loader=yaml.SafeLoader)
+yaml.add_constructor('!include', include, Loader=yaml.SafeLoader)
+yaml.add_constructor('!startstopfreq', startstopfreq, Loader=yaml.SafeLoader)
```

Comment:
That's an interesting namespace issue. 

I'm not sure about the globals, but I am kind of curious if it would make sense to have a "unified" yaml parser that we could import/use in any of the tools that are ingesting a YAML. That may make it easier to deal with the conflicts.